profiler/src/ProfilerEngine/Datadog.Profiler.Native/CollectorBase.h

```diff
@@ -161,6 +161,16 @@ class CollectorBase
         ProcessID pid;
         std::string appDomainName;
 
+        // check for null AppDomainId (garbage collection for example)
+        if (rawSample.AppDomainId == 0)
+        {
+            sample.SetAppDomainName("CLR");
+            // TODO: is pid mandatory?
+            sample.SetPid("0");
```

Comment:
why not setting the current process id ?

Reply:
The current implementation relies on a call to ICorProfilerInfo that requires the AppDomainID that we don't have